docs/beamrunner/beamrunner-3-sample.md

```diff
@@ -0,0 +1,172 @@
+
+```
+PCollection<KV<String,Double>> mergedReadings = deviceReadings
+        .apply("MergeReadings", Flatten.<KV<String,Double>>pCollections())
+        .apply("WindowReadings",
+                Window.<KV<String,Double>>into(
+                        FixedWindows.of(Duration.standardSeconds(options.getWindowSize()))));
+```
+
+The windowed data is validated with a ValidateReadings transform that produces two output streams, one of good readings and one of bad readings:
+
+```
+final TupleTag<KV<String,Double>> goodTag = new TupleTag<KV<String,Double>>(){ ... }
+final TupleTag<KV<String,Double>> badTag = new TupleTag<KV<String,Double>>(){ ... }
+final double badTempThreshold = options.getBadTempThreshold();
+
+PCollectionTuple validatedReadings = mergedReadings.apply("ValidateReadings",
+        ParDo.of(new DoFn<KV<String,Double>,KV<String,Double>>() {
+            private static final long serialVersionUID = 1L;
+
+            @ProcessElement
+            public void processElement(ProcessContext c) {
+                String device = c.element().getKey();
+                Double temp = c.element().getValue();
+                if (temp < badTempThreshold) {
+                    // Good reading, output and update distribution metrics
+                    c.output(c.element());
+                    ...
+                } else {
+                    // Bad reading, output to side output, update counters
+                    c.output(badTag, c.element());
+                    ...
+                }
+            }
+        }).withOutputTags(goodTag, TupleTagList.of(badTag)));
+
+
+```
+
+On the good readings, statistics are calculated in a GoodStats transform and then logged by a GoodLog transform:
+
+```
+validatedReadings
+        .get(goodTag)
+        .apply("GoodStats", Combine.<String, Double, Stats>perKey(new DeviceStatsFn()))
+        .apply("GoodLog", ParDo.of(new LogWindowFn<Stats>("Temperature device statistics")));
+```
+
+The bad readings are counted by a BadCount transform and logged by the BadLog transform:
+
+```
+validatedReadings
+        .get(badTag)
+        .apply("BadCount", Count.<String,Double>perKey())
+        .apply("BadLog", ParDo.of(new LogWindowFn<Long>("Temperature device error count")));
+```
+
+The structure of the application becomes clear after it is displayed in the Streams console after the application is run.
+
+## Running the `TemperatureSample` application
+
+1. Make sure that your Streaming Analytics service on Bluemix is running.
+2. Navigate to the `$STREAMS\_RUNNER\_HOME/samples` directory. The Streams Runner toolkit provides all necessary files. Assuming that all environment variables are set as described in Installing the toolkit and that $VCAP\_SERVICES has credentials in it named beam-service, you can launch the `TemperatureSample` application with the following command:
+
+   ```

```

Comment:
add
`cd $STREAMS_RUNNER_HOME/samples` to this command so you can copy and paste easily.